### PR TITLE
[feat]:Form control state #4

### PR DESCRIPTION
**Pull request by fazzaamiarso**
What is this PR about?  Adding states to `FormControl` 
-

Related issue number: #4 

Description of changes:
- Created client directory to organize client method.
- Created methods to get form values in client.
- Implement `isPristine` and `isValid` state to `FormControl`.

Tag a reviewer: @ayoayco 

Tasks:
- [x] I have ran the tests to make sure nothing is broken (see CONTRIBUTING.md)
- [x] I have mentioned the issue this PR is addressing

### CODE REVIEW

File: packages/astro-reactive-form/index.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
 import Form from './Form.astro';
 export default Form;
+export * from './core/form-control-types';
+export * from './client';
```
Comment (ayoayco) [inline packages/astro-reactive-form/index.ts:4]: I think these two lines can be removed. We don't need to expose all form-control-types outside the package for now, and the `client` is already exported via package.json

File: packages/astro-reactive-form/client/controls.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	inputEl.addEventListener('change', (e) => {
+		if (!(e.target instanceof HTMLInputElement)) return;
+		let value = e.target.value;
+		if (formControl.type === 'checkbox') {
+			value = formControl.value === 'checked' ? '' : 'checked';
+		}
+		formControl.setValue(value);
+		formControl.setIsPristine(false);
+	});
```
Comment (ayoayco) [inline packages/astro-reactive-form/client/controls.ts:45]: I'm thinking `setIsPristine` should be private to FormControl so users cannot do this themselves. So maybe we can set this up in the FormControl constructor? Let me know whether this works or if it creates problems.

File: packages/astro-reactive-form/components/Field.astro [hunk 3]
```diff
@@ -22,11 +22,13 @@ const { control } = Astro.props;
 		value={control.value}
 		checked={control.value === 'checked'}
 		placeholder={control.placeholder}
+		data-label={control.label}
+		data-label-position={control.labelPosition}
```
Comment (ayoayco) [inline packages/astro-reactive-form/components/Field.astro:26]: This is clever. 👍

**Review by ayoayco — CHANGES_REQUESTED** [review]
Almost perfect! 🚀 I just have some questions and suggestions for you to try. :)

File: packages/astro-reactive-form/client/controls.ts (hunk 2 above)
Comment (fazzaamiarso) [reply packages/astro-reactive-form/client/controls.ts:45]: I'm able to solve this using `proxy`.

So, I remove `setIsPristine` and replace it with `set isPristine (value) { this._isPristine = value }` in FormControl. After that, we just return a proxy of FormControl  in `getFormControl` that basically ignore any changes made using setter. 

```ts
//..omitted for brevity
formControl.setValue(value);
formControl.isPristine = false; //setter still works here
});

const controlProxy = new Proxy(formControl, {
          set() {
	      //prevent any setter to be able to work in client;
	      console.error('Setting this property manually is prohibited!');
	      return true;
          },
});
return controlProxy;
```

**Comment by ayoayco** [comment]
Great work!

**Pull request merged**